### PR TITLE
Feature/#7503 dcc request

### PR DESCRIPTION
**Pull request by pascal-brause**
## Description
Adds the HTTP call to receive the DCC itself. 

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7503

## Link to TechSpec
https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/test-certificate/docs/spec/dgc-test-certificate-client.md#api-for-dgc-test-certificate-components

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient+Configuration.swift [hunk 1]
```diff
@@ -32,7 +32,7 @@ extension HTTPClient {
 					requiresTrailingSlash: false
 				),
 				dcc: .init(
-					baseURL: environmentProvider.currentEnvironment().dccURL,
+					baseURL: environmentProvider.currentEnvironment().dgcURL,
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient+Configuration.swift]: dcc?
```suggestion
					baseURL: environmentProvider.currentEnvironment().dccURL,
```

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift [hunk 2]
```diff
@@ -526,6 +526,70 @@ final class HTTPClient: Client {
...
+							DCCResponse.self,
+							from: responseBody
+						)
+						completion(.success(decodedResponse))
+					} catch {
+						Log.error("Failed to decode response json", log: .api, error: error)
+						completion(.failure(.jsonError))
+					}
+				case 202:
+					Log.error("HTTP error code 202. DGC is pending.", log: .api)
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:566]: dcc?
```suggestion
					Log.error("HTTP error code 202. DCC is pending.", log: .api)
```

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDigitalCovid19CertificateTests.swift [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+final class HTTPClientGetDigitalCovid19CertificateTests: CWATestCase {
+	
+	func testGIVEN_RegistrationToken_WHEN_HappyCase_THEN_DCCResponseIsReturned() throws {
+		// GIVEN
+		let registrationToken = "someToken"
+		
+		let expectedResponse = DCCResponse(
+			dek: "someKey",
+			dcc: "someCOSE"
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDigitalCovid19CertificateTests.swift:17]: I find it irritating to see "dcc" assigned to something COSE-related. Until now I thought 'dcc' is something like 'digital covid certificate' and now _everything_ is dcc 🤯

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient+Configuration.swift (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient+Configuration.swift]: already fixed.

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDigitalCovid19CertificateTests.swift (hunk 3 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDigitalCovid19CertificateTests.swift:17]: just implemented like the tech spec wanted 🤷‍♂️

**Review by KaiTeuber — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDigitalCovid19CertificateTests.swift (hunk 3 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+GetDigitalCovid19CertificateTests.swift:17]: The Spec doesn't have to maintain this mess… 🙄

**Review by whiskey — APPROVED** [review]

**Pull request merged**